### PR TITLE
Add GitHub actions workflow to run unit tests

### PR DESCRIPTION
**Pull request by bbrks**
- Adds Go module file for library dependencies (excludes `./example` deps)
- Adds GitHub Actions unit test with `-race` workflow
- Fixes test race condition caught by `-race`

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+    branches:
+      - 'master'
+
+jobs:
+  test-race:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/setup-go@v2
+        with:
+          go-version: 1.17.5
```
Comment (bbrks) [inline .github/workflows/ci.yml:25]:
```suggestion
      - name: Setup Go Faster
        uses: WillAbides/setup-go-faster@v1.7.0
```

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+  pull_request:
+    branches:
+      - 'master'
+
+jobs:
+  test-race:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Setup Go Faster
+        uses: WillAbides/setup-go-faster@v1.7.0
```
Comment (bbrks) [inline .github/workflows/ci.yml:24]:
```suggestion
      - name: Setup Go Faster
        uses: WillAbides/setup-go-faster@v1.7.0
        with:
          go-version: *
```

File: go.mod [hunk 3]
```diff
@@ -0,0 +1,13 @@
+module github.com/couchbase/go-blip
+
+go 1.18
```
Comment (markspolakovs) [inline go.mod:3]: Will this cause issues if `go-blip` is imported in packages that have `go 1.17` like SG?

File: .github/workflows/ci.yml [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+        with:
+          go-version: *
```
Comment (bbrks) [inline .github/workflows/ci.yml:26]:
```suggestion
      - name: Setup Go Faster
        uses: WillAbides/setup-go-faster@v1.7.0
        with:
          go-version: ^1.17.5 # >=1.17.5, <2.0.0
```

File: go.mod (hunk 3 above)
Comment (markspolakovs) [reply go.mod:3]: Discussed offline - won't cause issues like build failures, but might be worth locking to 1.17 just to ensure that we don't accidentally use any 1.18 features before sync_gateway is upgraded to 1.18.
```suggestion
go 1.17
```

**Review by markspolakovs — APPROVED** [review]

File: go.mod (hunk 3 above)
Comment (bbrks) [reply go.mod:3]:
```suggestion
go 1.17
```

**Pull request merged**